samcli/commands/exceptions.py

```diff
@@ -121,3 +122,17 @@ class FailedArnParseException(UserException):
     """
     Exception class when `sam test-runner init` fails to extract a resource name/id from an ARN.
     """
+
+
+class InvalidTestRunnerTemplateException(UserException):
+    """
+    Exception raised when the customer passes a Test Runner CFN template to `sam test-runner run`
+    that has key resources duplicated or missing.
+    """
+
+
+class MissingTestRunnerTemplateException(UserException):
+    """
+    Exception raised when the customer attempts to run a testsuite with a Test Runner stack name that does not exist,
```

Comment:
During the first run the stack will not exist, but there should be a template supplied to create it. If a template is not supplied and there is no stack that already exists, we raise this exception. We keep the template as an optional parameter because it is only necessary for the first run, and any other updates the customer wishes to perform.